### PR TITLE
feat: add missing capabilities to deadline_test

### PR DESCRIPTION
**Pull request by locnnil**
## Summary
Add missing capabilities to deadline_test app: 
- add interface [mount-observe](https://snapcraft.io/docs/mount-observe-interface)
- add interface [system-trace](https://snapcraft.io/docs/system-trace-interface)

- Related to #8.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -25,6 +25,10 @@ platforms:
     build-for: [arm64]
 
 plugs:
+  sched-debugfs:
```
Comment (farshidtz) [inline snap/snapcraft.yaml:28]:
```suggestion
  system-scheduler-debugfs:
```
To make it a bit clearer that this is a system plug. Maybe even `system-files-scheduler-debugfs`, so the user can guess the plug?

Comment (locnnil) [reply snap/snapcraft.yaml:28]: But aren't most plugs system plugs? They provide capabilities to interact with the host system. I could rename it to `scheduler-debugfs` to make it clearer what it is about.

Comment (farshidtz) [reply snap/snapcraft.yaml]: Try if the following works:
```
sudo snap connect rt-tests:sched-debugfs
```

Comment (locnnil) [reply snap/snapcraft.yaml]: It works!

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -49,10 +49,12 @@ jobs:
...
         run: |
-          sudo snap connect rt-tests:process-control :process-control
-          sudo snap connect rt-tests:mount-observe :mount-observe
-          sudo snap connect rt-tests:system-trace :system-trace
-          sudo snap connect rt-tests:cpu-latency-dev rt-tests:dma-latency-dev
+          sudo snap connect rt-tests:process-control
+          sudo snap connect rt-tests:mount-observe
+          sudo snap connect rt-tests:system-trace
+          sudo snap connect rt-tests:scheduler-debugfs
+          #sudo snap connect rt-tests:cpu-latency-dev rt-tests:dma-latency-dev
```
Comment (farshidtz) [inline .github/workflows/build.yaml:56]: Why is this commented out?

Comment (locnnil) [reply .github/workflows/build.yaml:56]: It was before the fix-confinement branch has been synced to my current branch. I will put it back.

**Review by farshidtz — APPROVED** [review]

**Pull request merged**